src/getsetall.jl

```diff
@@ -0,0 +1,70 @@
+"""
+    getall(obj, optic)
+
+Extract all parts of `obj` that are selected by `optic`.
+Returns a flat tuple of values, or a vector if the selected parts contain vectors.
```

Comment:
```suggestion
Returns a flat `Tuple` of values, or a `Vector` if the selected parts contain vectors.
This function is experimental and we might change the precise output container in future.
```
For instance, we might want to return SVectors in certain situtations.